### PR TITLE
SALTO-5524: workflowV1 removal filter

### PR DESCRIPTION
**Pull request by shayc331**
_workflowV1 removal filter_

---

_Additional context for reviewer_
* removes workflowV1 instances when enableNewWorkflowAPI is enabled

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/jira-adapter/src/adapter.ts [hunk 1]
```diff
@@ -228,6 +229,7 @@ export const DEFAULT_FILTERS = [
   fieldNameFilter,
   workflowStructureFilter,
   workflowFilter,
+  workflowV1RemovalFilter,
```
Comment (IdoZakk) [inline packages/jira-adapter/src/adapter.ts:232]: Maybe do it before the other workflow filters, so we do not run for non reason on the old items?

File: packages/jira-adapter/src/filters/workflowV2/workflowV1_removal.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import _ from 'lodash'
+import { isInstanceElement, Element } from '@salto-io/adapter-api'
+import { FilterCreator } from '../../filter'
+import { WORKFLOW_TYPE_NAME } from '../../constants'
+
+const filter: FilterCreator = ({ config }) => ({
+  name: 'workflowV1RemovalFilter',
+  onFetch: async (elements: Element[]) => {
+    const { enableNewWorkflowAPI } = config.fetch
+    if (enableNewWorkflowAPI) {
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/workflowV2/workflowV1_removal.ts:25]: You can combine those lines

File: packages/jira-adapter/src/filters/workflowV2/workflowV1_removal.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      _.remove(elements, element => isInstanceElement(element) && element.elemID.typeName === WORKFLOW_TYPE_NAME)
+    }
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/workflowV2/workflowV1_removal.ts:27]: Maybe remove also the types while you are at it?

File: packages/jira-adapter/test/filters/workflowV2/workflowV1_removal.test.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    it('should remove workflowV1 instances when enableNewWorkflowAPI is true', async () => {
+      config.fetch.enableNewWorkflowAPI = true
+      filter = workflowV1RemovalFilter(
+        getFilterParams({
+          config,
+        }),
+      ) as typeof filter
+      await filter.onFetch(elements)
+      expect(elements).toHaveLength(3)
+      expect(elements).toEqual([workflowType, workflowV2Instance, notWorkflowInstance])
```
Comment (IdoZakk) [inline packages/jira-adapter/test/filters/workflowV2/workflowV1_removal.test.ts:56]: Nit: you can use `.toContainEqual` when the order of the items is not important

**Review by IdoZakk — APPROVED** [review]
🦸 
See my comments

File: packages/jira-adapter/test/filters/workflowV2/workflowV1_removal.test.ts (hunk 4 above)
Comment (shayc331) [reply packages/jira-adapter/test/filters/workflowV2/workflowV1_removal.test.ts:56]: nice!

**Pull request merged**